components/TARGET_PSA/spm/COMPONENT_SPE/spm_internal.h

```diff
@@ -0,0 +1,237 @@
+    uint32_t partition_count; /* Number of Secure Partitions in the system.*/
+    psa_handle_manager_t channels_handle_mgr;
+    psa_handle_manager_t messages_handle_mgr;
+    osMemoryPoolId_t channel_mem_pool; /* Channel memory pool identifier.*/
+    osMemoryPoolId_t active_messages_mem_pool; /* Channel memory pool identifier.*/
+} spm_db_t;
+
+/*
+ * Returns a pointer to the currently active secure partition or NULL in case called from NSPE.
+ */
+spm_partition_t *get_active_partition(void);
+
+/*
+ * Return an array of memory regions used by a given partition.
+ *
+ * @param[in] partition_id - a partition ID to find memory regions for, if MEM_PARTITIONS_ALL then
+ *                           memory regions for all the partitions are returned
+ * @param[out] region_count - will be set to the number of memory regions returned
+ */
+const mem_region_t *get_mem_regions(int32_t partition_id, uint32_t *region_count);
+
+// Platform dependent APIs
+
+/**
+ * Validates a memory block is accessable from a specific partition
+ *
+ * @param[in] ptr pointer to the beggining of the memory block.
+ * @param[in] size size of the memory block in bytes.
+ * @param[in] accessing_partition which partition is trying to access the memory.
+ * @return true if the entire memory block is accessable from given partition.
+ */
+bool is_buffer_accessible(const void *ptr, size_t size, spm_partition_t *accessing_partition);
+
+/**
+ * Alerts NSPE that a proccess (connect or call) has ended.
+ *
+ * @param[in] completion_sem_id semaphore id in NSPE.
+ */
+void nspe_done(osSemaphoreId_t completion_sem_id);
+
+/*
+ * Validates parameters sent from caller and queues a connect message on the correct ROT_SRV.
+ *
+ * @param[in] sid - desired RoT service ID
+ * @param[in] msg - pointer to connect message struct
+ */
+void psa_connect_async(uint32_t sid, spm_pending_connect_msg_t *msg);
+
+/*
+ * Validates parameters sent from caller and queues a call message on the correct ROT_SRV.
+ *
+ * @param[in] handle - channel handle for the connection
+ * @param[in] msg - pointer to call message struct
+ */
+void psa_call_async(psa_handle_t handle, spm_pending_call_msg_t *msg);
+
+/*
+ * Validates parameters sent from caller and queues a disconnect message on the correct ROT_SRV.
+ *
+ * @param[in] handle - handle of channel to close
+ * @param[in] msg - pointer to close message struct
+ */
+void psa_close_async(psa_handle_t handle, spm_pending_close_msg_t *msg);
+
+
+/*
+ * Validates IOvecs.
+ *
+ * @param[in] in_vec - psa_invec_t array
+ * @param[in] in_len - number of elements in in_vec
+ * @param[in] out_vec - psa_outvec_t array
+ * @param[in] out_len - number of elements in out_vec
+*/
+void validate_iovec(
+    const void *in_vec,
+    const uint32_t in_len,
+    const void *out_vec,
+    const uint32_t out_len
+);
+
+void channel_state_switch(uint8_t *current_state, uint8_t expected_state, uint8_t new_state);
```

Comment:
these 2 functions do not have doxygen, the rest of the functions here have?

Reply:
This is confusing.
This is an internal API.
AFAIU, it should not be Doxygen documented. Of course it should have normal code comments.
My bottom line is: This file comments should be be Doxygen style.